sdk/helper/aliasmetadata/alias_metadata.go

```diff
@@ -0,0 +1,212 @@
+	// schema.
+	FieldName string
+
+	// Default is a list of the default fields that should
+	// be included if a user sends "default" in their list
+	// of desired fields. These fields should all have a
+	// low rate of change because each change can incur a
+	// write to storage.
+	Default []string
+
+	// AvailableToAdd is a list of fields not included by
+	// default, that the user may include.
+	AvailableToAdd []string
+}
+
+func (f *Fields) all() []string {
+	return append(f.Default, f.AvailableToAdd...)
+}
+
+// FieldSchema takes the default and additionally available
+// fields, and uses them to generate a verbose description
+// regarding how to use the "alias_metadata" field.
+func FieldSchema(fields *Fields) *framework.FieldSchema {
+	return &framework.FieldSchema{
+		Type:        framework.TypeCommaStringSlice,
+		Description: description(fields),
+		DisplayAttrs: &framework.DisplayAttributes{
+			Name:  fields.FieldName,
+			Value: "default,field1,field2",
+		},
+		Default: []string{"default"},
+	}
+}
+
+func NewHandler(fields *Fields) *Handler {
+	return &Handler{
+		fields: fields,
+	}
+}
+
+type Handler struct {
+	// aliasMetadata is an explicit list of all the user's configured
+	// fields that are being added to alias metadata. It will never
+	// include the "default" parameter, and instead includes the actual
+	// fields behind "default", if selected. If it has never been set
+	// or only the default fields are desired, it is nil.
+	aliasMetadata []string
+
+	// fields is a list of the configured default and available
+	// fields. It's intentionally not jsonified.
+	fields *Fields
+}
+
+// AliasMetadata is intended to be used on config reads.
+// It gets an explicit list of all the user's configured
+// fields that are being added to alias metadata. It will never
+// include the "default" parameter, and instead includes the actual
+// fields behind "default", if selected.
+func (h *Handler) AliasMetadata() []string {
+	if h.aliasMetadata == nil {
+		return h.fields.Default
+	}
+	return h.aliasMetadata
+}
+
+// ParseAliasMetadata is intended to be used on config create/update.
+// It takes a user's selected fields (or lack thereof),
+// converts it to a list of explicit fields, and adds it to the Handler
+// for later storage.
+func (h *Handler) ParseAliasMetadata(data *framework.FieldData) error {
+	userProvidedRaw, ok := data.GetOk(h.fields.FieldName)
+	if !ok {
+		// Nothing further to do here.
+		return nil
+	}
+	userProvided, ok := userProvidedRaw.([]string)
+	if !ok {
+		return fmt.Errorf("%s is an unexpected type of %T", userProvidedRaw, userProvidedRaw)
+	}
+
+	// If the only field the user has chosen was the default field,
+	// we don't store anything so we won't have to do a storage
+	// migration if the default changes.
+	if len(userProvided) == 1 && userProvided[0] == "default" {
+		h.aliasMetadata = nil
+		return nil
+	}
+
+	userProvided = h.expandDefaultField(userProvided)
```

Comment:
In our design doc we described "default" as special in that if provided, it should be the only thing provided, otherwise it is an error. If we expand defaults, it may be confusing since:
```
alias_metadata="default" -- Results in a set of defaults that floats with any Vault changes

alias_metadata="default, abc" -- Results in the current defaults being stored.
```
The behavior of "default"  becomes more difficult to explain as compared to: "the parameter should contain a list of metadata fields, or 'default'"

Reply:
Ah! I missed that, fixed in https://github.com/hashicorp/vault/pull/8783/commits/a7987073e3b3b9ba96c23a8d9f95de99914d2a49 and test updated.